python/graphstorm/distiller/distiller.py

```diff
@@ -0,0 +1,351 @@
+        save_model_frequency,
+        eval_frequency,
+        max_global_step,
+        on_cpu=False,
+    ):
+        """ Distill function.
+        Each worker initializes student model, data provider,
+        and start training.
+
+        Parameters
+        ----------
+        lm_name : str
+            Model name for Transformer-based student model.
+        pre_trained_name : str
+            Name for pre-trained model weights.
+        gnn_embed_dim : int
+            Size of GNN embeddings.
+        data_dir : str
+            Directory for distillation data.
+        batch_size : int
+            Batch size for distillation.
+        max_seq_len : int
+            Maximum sequence length.
+        distill_lr : float
+            Learning rate for distillation.
+        saved_path : str
+            Path to save the model.
+        save_model_frequency : int,
+            Interval to save the model checkpoint.
+        eval_frequency : int,
+            Interval for evaluation.
+        max_global_step : int,
+            Maximum steps for distillation training.
+        on_cpu : bool
+            Whether the distillation will be conducted on cpu.
+        """
+        self.student = GSDistilledModel(lm_name=lm_name, pre_trained_name=pre_trained_name)
+        dataloader_generator = DataloaderGenerator(tokenizer=self.student.tokenizer, 
+            max_seq_len = max_seq_len,
+            device=self.device, 
+            batch_size=batch_size,
+        )
+        train_data_provider = DataManager(
+            dataloader_generator,
+            dataset_path=os.path.join(data_dir, 'train'),
+            local_rank=self.rank,
+            world_size=th.distributed.get_world_size(),
+            is_train=True,
+        )
+        eval_data_provider = DataManager(
+            dataloader_generator,
+            dataset_path=os.path.join(data_dir, 'val'),
+            local_rank=self.rank,
+            world_size=th.distributed.get_world_size(),
+            is_train=False,
+        )
+
+        # get GNN embed dim
+        dataset_iterator = eval_data_provider.get_iterator()
+        if not dataset_iterator:
+            raise RuntimeError("No validation data") 
+        batch = next(iter(dataset_iterator))
+        gnn_embed_dim = batch["labels"].shape[1]
+        self.student.init_proj_layer(gnn_embed_dim=gnn_embed_dim)
+
+        # TODO (HZ): add flexibility to specify different optimizers
+        optimizer = th.optim.Adam(self.student.parameters(), lr=distill_lr)
+        self.student.to(self.device)
+        student = DistributedDataParallel(self.student, device_ids=None if on_cpu else [self.device],
+                                        output_device=None if on_cpu else self.device)
+
+        index = 0
+        global_step = 0
+        while True:
+            th.distributed.barrier()
+            print (f"Train {index + 1}-th shard by trainer {self.rank}")
+            complete, global_step = self.train_shard(
```

Comment:
how is the global step different from local step?

Reply:
change to distill step by new commit though.